src/workloads/execution/FilterWithComplexLogicalExpression.yml

```diff
@@ -426,6 +434,535 @@ Actors:
               ]
             }
 
+# Clean up the data from the previous phase.
+- Name: ClearCollection
+  Type: CrudActor
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [12]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Collection: *Collection
+        Operations:
+        - OperationName: drop
+
+- Name: QuiesceCollectionDrop
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [13]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+
+# Pair of loaders, the first of which inserts documents with the path "a.c.d", the second inserts
+# documents with the path "b.c.d".
+- Name: InsertDocumentsWithTargetPath
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [14]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        Threads: 1
+        CollectionCount: 1
+        DocumentCount: *HalfDocumentCount
+        BatchSize: 1000
+        Document:
+          c: 1
+          d: 1
+          a: { c: { d: 1 } }
+
+- Name: InsertDocumentsWithIgnoredPath
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [14]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        Threads: 1
+        CollectionCount: 1
+        DocumentCount: *HalfDocumentCount
+        BatchSize: 1000
+        Document:
+          c: 1
+          d: 1
+          b: { c: { d: 1 } }
+
+- Name: QuiesceDocumentsWithTargetAndMissingPaths
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [15]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+
+# Both queries below will only target "a.c.d", skipping the documents which have "b.c.d".
+- Name: AggregationExpressionTargetPath
+  Type: CrudActor
+  Database: *Database
+  Threads: *Threads
+  Phases:
+    OnlyActiveInPhases:
+      Active: [16]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: *Repeat
+        Collection: *Collection
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {$expr: {$eq: ["a.c.d", 0]}}
+
+- Name: MatchExpressionTargetPath
+  Type: CrudActor
+  Database: *Database
+  Threads: *Threads
+  Phases:
+    OnlyActiveInPhases:
+      Active: [17]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: *Repeat
+        Collection: *Collection
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {"a.c.d": 0}
```

Comment:
Could you add a case that the entire path is missing, e.g. `a.c.e`?

Reply:
The later one, suffix doesn't exist.

Reply:
Ok, done.